lib/JsonpMainTemplatePlugin.js

```diff
@@ -2,205 +2,192 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-var Template = require("./Template");
+"use strict";
 
-function JsonpMainTemplatePlugin() {}
-module.exports = JsonpMainTemplatePlugin;
+const Template = require("./Template");
+
+class JsonpMainTemplatePlugin {
 
-JsonpMainTemplatePlugin.prototype.constructor = JsonpMainTemplatePlugin;
-JsonpMainTemplatePlugin.prototype.apply = function(mainTemplate) {
-	mainTemplate.plugin("local-vars", function(source, chunk) {
-		if(chunk.chunks.length > 0) {
+	apply(mainTemplate) {
+		mainTemplate.plugin("local-vars", function(source, chunk) {
+			if(chunk.chunks.length > 0) {
+				return this.asString([
+					source,
+					"",
+					"// objects to store loaded and loading chunks",
+					"var installedChunks = {",
+					this.indent(
+						chunk.ids.map(id => `${id}: 0`).join(",\n")
+					),
+					"};"
+				]);
+			}
+			return source;
+		});
+		mainTemplate.plugin("jsonp-script", function(_, chunk, hash) {
+			const chunkFilename = this.outputOptions.chunkFilename;
+			const chunkMaps = chunk.getChunkMaps();
+			const crossOriginLoading = this.outputOptions.crossOriginLoading;
+			const chunkLoadTimeout = this.outputOptions.chunkLoadTimeout || 120000;
 			return this.asString([
-				source,
-				"",
-				"// objects to store loaded and loading chunks",
-				"var installedChunks = {",
-				this.indent(
-					chunk.ids.map(function(id) {
-						return id + ": 0";
-					}).join(",\n")
-				),
-				"};"
+				"var script = document.createElement('script');",
+				"script.type = 'text/javascript';",
+				"script.charset = 'utf-8';",
+				"script.async = true;",
+				`script.timeout = ${chunkLoadTimeout};`,
+				crossOriginLoading ? `script.crossOrigin = '${crossOriginLoading}';` : "",
+				`if (${this.requireFn}.nc) {`,
+				this.indent(`script.setAttribute("nonce", ${this.requireFn}.nc);`),
+				"}",
+				`script.src = ${this.requireFn}.p + ${this.applyPluginsWaterfall("asset-path", JSON.stringify(chunkFilename), {
+					hash: `" + ${this.renderCurrentHashCode(hash)} + "`,
+					hashWithLength: length => `" + ${this.renderCurrentHashCode(hash, length)} + "`,
+					chunk: {
+						id: "\" + chunkId + \"",
```

Comment:
not possible, this is just a string. This is the whole point of bringing in single quotes.